### PR TITLE
[Application] Fix deploying from image

### PR DESCRIPTION
**Pull request by TomerShor**
Deploying an application runtime from image was impossible because the reverse-proxy source code was always populated at first deployment.

Fix is to set the image on the status before the deployment, so the reverse proxy source code won't be set on the function spec.

Furthermore, make sure that if `spec.image` is set, we do not enrich the source code in the backend.

### CODE REVIEW

File: mlrun/runtimes/nuclio/application/application.py [hunk 1]
```diff
@@ -367,13 +367,21 @@ def with_source_archive(
         )
 
     def from_image(self, image):
+        """
+        Deploy the function from an existing image
```
Comment (alonmr) [inline mlrun/runtimes/nuclio/application/application.py]:
```suggestion
        Deploy the function with an existing nuclio processor image. This does not applies for the reverse proxy and not the application image.
```

File: server/api/crud/runtimes/nuclio/function.py [hunk 2]
```diff
@@ -347,13 +347,15 @@ def _compile_function_config(
...
+    is_nuclio_image_set = function.spec.config.get("spec.image") is not None
+
     # if the user code is given explicitly or from a source, we need to set the handler and relevant attributes
     if (
         function.spec.base_spec
         or function.spec.build.functionSourceCode
         or function.spec.build.source
         or function.kind == mlrun.runtimes.RuntimeKinds.serving  # serving can be empty
-    ):
+    ) and not is_nuclio_image_set:
```
Comment (alonmr) [inline server/api/crud/runtimes/nuclio/function.py]: can you explain this? The else clause seems to be weird. why would we want to build file if a nuclio image is set?

**Review by alonmr — COMMENTED** [review]
thanks <3 minor comment and question

File: mlrun/runtimes/nuclio/application/application.py (hunk 1 above)
Comment (TomerShor) [reply mlrun/runtimes/nuclio/application/application.py]: This **does** apply for the reverse proxy...

File: server/api/crud/runtimes/nuclio/function.py (hunk 2 above)
Comment (TomerShor) [reply server/api/crud/runtimes/nuclio/function.py]: You're right, I don't want to enter the "else" clause.
The actual meaning was to not call `_set_source_code_and_handler`, so I'll move the condition there.

**Review by alonmr — APPROVED** [review]

**Pull request merged**